### PR TITLE
Update APIKey struct

### PR DESCRIPTION
**Pull request by susanpann**
[sc-43458]
Fixes https://github.com/OctopusDeploy/Issues/issues/8434

Created different types for the POST response and GET response as the "APIKey" field is different
- Post response:
Screenshot 2023-10-30 at 1 10 52 pm

- GET response:
apikeys-get-response

### CODE REVIEW

File: pkg/users/api_key.go [hunk 1]
```diff
@@ -6,8 +6,8 @@ import (
 	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/resources"
 )
 
-// APIKey represents an API key.
-type APIKey struct {
+// CreateAPIKey represents an API key when returned from creating a new API key (POST)
```
Comment (veochen-octopus) [inline pkg/users/api_key.go:9]: Looks like this is used as both the request and the response

```suggestion
// CreateAPIKey represents an API key when creating a new API key (POST)
```
which makes `CreateAPIKey` kind of a weird name but I don't have any good suggestions so feel free to leave it.

File: test/e2e/api_keys_test.go [hunk 2]
```diff
@@ -1,86 +1,96 @@
...
-// 		assert.NotNil(t, key)
-// 	}
+	apiKeys, err := client.APIKeys.GetByUserID(user.ID)
+	require.NotNil(t, apiKeys)
+	require.NoError(t, err)
 
-// 	err = octopusClient.Users.DeleteByID(user.id)
-// 	assert.NoError(t, err)
-// }
+	CleanupTestUser(t, client, user)
```
Comment (veochen-octopus) [inline test/e2e/api_keys_test.go:55]: Minor: I think wrapping this in `Cleanup` is slightly better as it makes sure this func is always called even if the test panics.

```suggestion
	t.Cleanup(func() { CleanupTestUser(t, client, user) })
```

**Review by veochen-octopus — APPROVED** [review]
The changes and the tests look good. Just a couple minor comments for consideration.

File: pkg/users/api_key.go (hunk 1 above)
Comment (susanpann) [reply pkg/users/api_key.go]: will keep the name for now as I can't think of anything else 😅

**Pull request merged**